crates/solvers/src/domain/mod.rs

```diff
@@ -3,6 +3,14 @@
 pub mod auction;
 pub mod baseline;
 pub mod eth;
+pub mod legacy;
 pub mod liquidity;
 pub mod order;
 pub mod solution;
+
+pub trait Solver: Send + Sync {
+    fn solve(
+        &self,
+        auction: auction::Auction,
+    ) -> futures::future::BoxFuture<Vec<solution::Solution>>;
+}
```

Comment:
Since the `solvers` crate knows all possible `Solver` implementations before hand, then having an `enum` makes you not lose type information:

```
pub enum Solver {
    Baseline(baseline::Baseline),
    Legacy(legacy::Legacy),
}
```

This way, we can `match` and figure out what solver kind is being used instead of needing to change the `Solver` trait to be able to extract that information, or use downcasting.

An argument for a trait would be that we have potentially external implementations of `trait Solver` that cannot be known ahead of time, but this shouldn't be the case here.

I don't feel strongly about this point, so I'm fine as is, but we've had this discussion before on trait objects vs enums, so I think it is worth highlighting here.

Reply:
Hmm, I still don't really see the appeal of using enums instead of `dyn Trait` for these kinds of things. It seems to me if we care about the underlying type implementing the `dyn Trait` (for example to handle some weird/specific behavior) then the trait isn't designed correctly.

Reply:
I don't see anything necessarily wrong with a trait here, just that we don't _need_ a trait, since all `Solver` implementations are known upfront: `baseline::Baseline | legacy::Legacy`.

As I said, don't feel strongly either way, just thought it was worth highlighting that we decided to use a trait here.

Reply:
> It seems like a single solver should return a single solution.

The idea is that solver engines return multiple solution proposals and that the `driver` would simulate each of them and pick the best one for solver competition.

Reply:
I think the argument that we wouldn't have to use `BoxFuture` has merit (although the trade-off will be the boilerplate code on the enum itself).
Returning multiple solutions was introduced by the `baseline` solver implementation and is supposed to allow us to compute settlements for individual orders and merge them afterwards.
@nlordell since you have loads of branches depending on this. Should I change it now or should we do it on-top of your changes to avoid possible merge conflicts?

Reply:
> @nlordell since you have loads of branches depending on this. Should I change it now or should we do it on-top of your changes to avoid possible merge conflicts?

I would change it now. This should be fairly easy merge conflicts to resolve.